mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageResultIterator.java

```diff
@@ -154,7 +159,21 @@ private void readBatch() throws MailboxException {
             range = MessageRange.range(cursor, to);
             break;
         }
-        next = mapper.findInMailbox(mailbox, range, ftype, batchSize);
+        next = mapper.findInMailbox(mailbox, range, ftype, batchSizeFromFetchType(ftype));
+    }
+
+    private int batchSizeFromFetchType(FetchType fetchType) {

```

Comment:
The _BatchSizes_ is a store of different values, then the usage is delegated to the customer, IMHO.
You are doing the same thing when you have a stream and then doing some filter then some map.